etc/profile-a-l/bcompare.profile

```diff
@@ -0,0 +1,69 @@
+# Firejail profile for Beyond Compare by Scooter Software
+# Description: directory and file compare utility
+# Disables the network, which only impacts checking for updates.
+# This file is overwritten after every install/update
+
+# Persistent local customizations
+include bcompare.local
+# Persistent global definitions
+include globals.local
+
+# In case the user decides to include disable-programs.inc, still allow KDE's Gwenview to view images via right click -> Open With -> Associated Application
+noblacklist ${HOME}/.config/gwenviewrc
+
+include disable-common.inc
+include disable-devel.inc
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-passwdmgr.inc
+# Allow comparing e.g. Firefox profile files, so don't disable access to programs
+# include disable-programs.inc
+# Uncommenting this breaks launch
+# include disable-shell.inc
```

Comment:
```suggestion
```

no need to keep such lines (in general it's often enough to `include allow-bin-sh.inc`)

Reply:
I've kept them because to me it was surprising to realize that `xdg` was related to `${DOCUMENTS}`, and I thought others who base the profile on this one might benefit from the comment. For the same reason, I added potentially superfluous comments like

```
# Might break Pulse Audio
#machine-id
```

or

```
# Allow associated applications to play sound files
#nosound
```

Reply:
That seems a bit arbitrary... disable all shells, but then enable back `bash`, `dash`, and `sh`, but not the much more popular `zsh`. Why bother disabling shells then?

Reply:
zsh is popular as interactive shell yes, but not to write script in it as you can not assume it's installed on system. Usually you write you scripts for `/bin/sh` which should be present on any POSIX compatible system or `/bin/bash` which should be present on the most GNU/Linux systems. `allow-bin-sh.inc` does exactly what it names says, it allows `/bin/sh`. `/bin/sh` is a symlink to `/bin/bash` on the most system resp. `/bin/dash` on Debian/Ubuntu. Therefore they need a `noblacklist` too.